code/game/objects/items/toys.dm

```diff
@@ -127,6 +127,64 @@
 	throw_range = 7
 	force = 0
 	var/random_color = TRUE
+	var/world_state
+	var/storage_state
+	var/current_color
+
+/obj/item/toy/balloon/long
+	name = "long balloon"
+	desc = "A perfect balloon to contort into goofy forms. Sealed with a mechanical bluespace wrap so it remains floating no matter what."
+	icon_state = "balloon_long"
+	inhand_icon_state = "balloon"
+	w_class = WEIGHT_CLASS_NORMAL
+
+/obj/item/toy/balloon/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	if(isturf(loc))
+		icon_state = "[world_state]"
+	else
+		icon_state = "[storage_state]"
+	update_appearance()
+
+/obj/item/toy/balloon/long/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(!istype(attacking_item, /obj/item/toy/balloon/long) || !HAS_TRAIT(user, TRAIT_BALLOON_SUTRA))
+		return ..()
+
+	var/obj/item/toy/balloon/long/hit_by = attacking_item
+	if(hit_by.current_color == current_color)
+		to_chat(user, span_warning("You must use balloons of different colours to do that!"))
+		return ..()
+	visible_message(
+		span_notice("[user.name] starts contorting up a balloon animal!"),
+		blind_message = span_hear("You hear balloons being contorted."),
+		vision_distance = 3,
+		ignored_mobs = user,
+	)
+	var/list/balloon_combos = list(
```

Comment:
Move this list into the object properties and add `static`, we don't want to recreate it every time this proc gets run.

Reply:
what do you mean?

Reply:
Move this list definition from inside this proc to being a property of the balloon object.
And make it `var/static/list/balloon_combos`